1_wqp_pull/src/wqp_inventory.R

```diff
@@ -171,27 +187,27 @@ summarize_wqp_inventory <- function(inv_ind, out_file) {
 #'   observations for that site/variable in WQP and the unique task identifier
 #'   that partitions site/variables into WQP pulls.
 partition_wqp_inventory <- function(partitions_ind, wqp_pull_params, inventory_ind, wqp_partition_cfg, pull_date) {
-  
+
   # Read in the inventory & config
   wqp_inventory <- feather::read_feather(sc_retrieve(inventory_ind, 'getters.yml'))
   wqp_partition_config <- wqp_partition_cfg
-  
+
   # filter out site types that are not of interest
-  
+
   wqp_inventory <- wqp_inventory %>%
     filter(!(ResolvedMonitoringLocationTypeName %in% wqp_pull_params$DropLocationTypeName))
-  
+
   # filter out bad org names
   # these orgs cause "Frequest failed [400]" errors
   bad_orgs <- grep(' |\\.|/', wqp_inventory$OrganizationIdentifier, value = TRUE)
   bad_orgs_sites <- filter(wqp_inventory, OrganizationIdentifier %in% bad_orgs)
```

Comment:
Rasha - can you find one of these "bad_orgs" and try to manually (that is, outside of the pipeline) pull date using `readWQPdata`? I want to be sure we still can't pull these.

Reply:
First, I pulled the WQP Stream data for 2018, then I identified "bad sites" using the chunk of code above. Afterwards, I pulled 2018 WQP data for the county where the "bad sites" is located. Then, I checked if the "bad sites" was pulled with the county data.

Lastly, I also experimented with pulling nation-wide stream temperature data pull for one day.

**R Code:**
```
# Retirivng the wqp streams data for the year 2018
wqp_stream2018_data <- readWQPdata(siteType = 'Stream', 
                                   sampleMedia = c('water', 'Water'),
                                   characteristicName = all_vars,
                                   startDateLo = '2018-01-01', startDateHi = '2018-12-31')

# grep search for matches to argument pattern within each element of a character vector
bad_orgs <- grep(' |\\.|/', wqp_stream2018_data$OrganizationIdentifier, value = TRUE)
bad_orgs_sites <- filter(wqp_stream2018_data, OrganizationIdentifier %in% bad_orgs)
# getting the unique bad site
bad_site_list <- unique(bad_orgs_sites$OrganizationIdentifier)

# trying to get the data for the bad site
# from 2018 data we identified a bad site "ALABAMACOUSHATTATRIBE.TX_WQX",
# Using google we idenified this site in  Polk County, Texas.
# now we will retrive te state and county code to pull all temperature data for 
# that county during 2018. then, check if that site is returned with that county data. 
stateCd # state_code for TX = 48 
counties <- countyCd # plok county = 373

polk_TX_data <- readWQPdata( siteType = sites_type,
                             #siteid = "ALABAMACOUSHATTATRIBE.TX_WQX", 
                             sampleMedia = c('water', 'Water'),
                             characteristicName = all_vars,
                             statecode="TX", countycode="373",
                             startDateLo = '2018-01-01', startDateHi = '2018-12-31')

# checking if bad site exists in polk-county,TX WQP data that we retrieved.
bad_site_check <- polk_TX_data %>%
  filter(OrganizationIdentifier %in% bad_site_list)

# pulling nation-wide data for one day of temperature data.
tic('test_1_day')
one_day_temp_day <- readWQPdata(siteType = 'Stream',
                                characteristicName = all_vars,
                                startDateLo = '2018-01-01', 
                                startDateHi = '2018-01-01')
toc()
```

** R Results:**
- Bad site checking
bad_site_list
"ALABAMACOUSHATTATRIBE.TX_WQX"

- Time of the nation-wide data for one day of temperature data
test_1_day: 0.45 sec elapsed

**Findings:**

The bad site "ALABAMACOUSHATTATRIBE.TX_WQX" was found in the Polk county, TX temperature data during 2018. 
It took 0.45 seconds to pull nation-wide stream temperature data pull for one day..